src/test/java/com/solvd/laba/steps/OrderSteps.java

```
@@ -9,66 +16,51 @@
 import com.solvd.laba.utils.MyBatisUtil;
 import com.solvd.laba.web.pages.*;
 
-import io.cucumber.java.en.*;
-import org.apache.ibatis.session.SqlSession;
-import org.openqa.selenium.WebDriver;
-
-import java.util.List;
-
-import static org.testng.Assert.assertTrue;
-
-public class OrderSteps {
+public class OrderSteps extends AbstractTest {
 
     private User user;
     private List<Order> orders;
     private SqlSession session;
-
-    private WebDriver driver;
+    private ProductsPage productsPage;
+    private CartPage cartPage;
+    private CheckoutCompletePage checkoutCompletePage;
 
     @Given("user {string} from DB logs in")
     public void userLogsInFromDb(String username) {
         session = MyBatisUtil.getSessionFactory().openSession();
```

Session is opened here but not guaranteed to be closed (only closed in @Then). This is risky in case a step fails early.
Wrap DB access in try (SqlSession session = …) to ensure safe closing